### PR TITLE
Implement a producer-consumer threading model with the Inara Service.

### PR DESCRIPTION
**Pull request by Tkael**
Ref. https://docs.microsoft.com/en-us/dotnet/standard/collections/thread-safe/blockingcollection-overview

### CODE REVIEW

File: EddiInaraService/InaraService.cs [hunk 1]
```diff
@@ -240,12 +237,15 @@ private bool checkAPIcredentialsOk(InaraConfiguration inaraConfiguration)
         public void SendQueuedAPIEvents(bool eddiIsBeta)
         {
             List<InaraAPIEvent> queue = new List<InaraAPIEvent>();
-            while (queuedAPIEvents.TryDequeue(out InaraAPIEvent pendingEvent))
+            // The `GetConsumingEnumerable` method blocks the thread while the underlying collection is empty
+            // If we haven't extracted events to send to Inara, this will wait / pause background sync until `queuedAPIEvents` is no longer empty.
+            foreach (var pendingEvent in queuedAPIEvents.GetConsumingEnumerable())
             {
                 queue.Add(pendingEvent);
+                if (queue.Count > 0 && queuedAPIEvents.Count == 0) { break; }
```
Comment (richardbuckle) [inline EddiInaraService/InaraService.cs:245]: Sorry, I don't quite follow why this has moved inside the foreach loop?

Comment (Tkael) [reply EddiInaraService/InaraService.cs:245]: The ForEach loop stops enumerating when the underlying collection is empty. Using a break here allows us to process the events extracted from collection (if there are any). If no events are extracted, the thread blocks.

Comment (Tkael) [reply EddiInaraService/InaraService.cs:245]: Stated more simply: this unblocks the thread when we have work to do.

Comment (Tkael) [reply EddiInaraService/InaraService.cs:245]: More information about the ForEach loop and `GetConsumingEnumerable()` method applied to a Blocking Collection: https://docs.microsoft.com/en-us/dotnet/standard/collections/thread-safe/how-to-use-foreach-to-remove

**Comment by richardbuckle** [comment]
Tests OK.

**Pull request merged**